### PR TITLE
Length Limit for Several Fields

### LINKED ISSUE #128: Parse more than 1 whitespaces in between input value to be 1 whitespace
Currently, it is possible to add an applicant `John Doe` and `John`       `Doe` as two different applicants.
We should edit each of the parsers to remove the whitespace in between the name parameters before checking for duplicate applicants.

### LINKED ISSUE #139: Set maximum character limit for every field given by the user
As a precaution, we need to set character limits for each field to ensure that the data input is valid and that every field is of an appropriate length to be shown in the UI.

Below are the specified limits that we have agreed on:
- Name: 50
- Address: 100
- Phone: 20
- Job ID: 10
- Job Title: 150
- Major: 50
- University: 100
- Tag: 30 per tag

### PR DESCRIPTION
**Pull request by cliftonfelix**
Resolves #139
Resolves #128 

- Value will be checked for length first before `REGEX`
- Added tests for `ParserUtil`
- Added tests for `isWithinLengthLimit`

Field Length Limit
- `Name` 50 chars
- `Email` 50 chars
- `Phone` 20 chars
- `University` 100 chars
- `Major` 50 chars
- `Address` 100 chars
- `Id` 10 chars
- `Title` 100 chars
- `Tag` 30 chars

### CODE REVIEW

File: docs/UserGuide.md [hunk 1]
```diff
@@ -36,6 +36,18 @@ InternConnect is a **desktop app for managing internship applicants, optimized f
 
 --------------------------------------------------------------------------------------------------------------------
 
+## Fields
```
Comment (cliftonfelix) [inline docs/UserGuide.md:39]: We put KIV first

**Review by JasonCP14 — COMMENTED** [review]
Maybe you can add individual tests for each of the Class.isWithinLengthLImit()

**Comment by cliftonfelix** [comment]
> Maybe you can add individual tests for each of the Class.isWithinLengthLImit()

We don't need to as ParserUtilTest already covers this

**Review by JasonCP14 — APPROVED** [review]
LGTM

**Review by lulucopter — APPROVED** [review]
Overall LGTM

**Pull request merged**